instrumentation/rspec/lib/opentelemetry/instrumentation/rspec/formatter.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry-api'
+require_relative './version'
+
+module OpenTelemetry
+  module Instrumentation
+    module RSpec
+      # An RSpec Formatter that outputs Otel spans
+      class Formatter
+        attr_reader :output
+
+        ::RSpec::Core::Formatters.register self, :example_started, :example_finished, :example_group_started, :example_group_finished, :start, :stop
+
+        @clock = Time.method(:now)
+
+        def self.current_timestamp
+          @clock.call
+        end
+
+        def initialize(output = StringIO.new, tracer_provider = OpenTelemetry.tracer_provider)
+          @spans_and_tokens = []
+          @output = output
+          @tracer_provider = tracer_provider
+        end
+
+        def tracer
+          @tracer ||= @tracer_provider.tracer('OpenTelemetry::Instrumentation::RSpec', OpenTelemetry::Instrumentation::RSpec::VERSION)
+        end
+
+        def current_timestamp
+          self.class.current_timestamp
+        end
+
+        def start(notification)
+          span = tracer.start_span('RSpec suite', start_timestamp: current_timestamp)
+          track_span(span)
+        end
+
+        def stop(notification)
+          pop_and_finalize_span
+        end
+
+        def example_group_started(notification)
+          description = notification.group.description
+          span = tracer.start_span(description, start_timestamp: current_timestamp)
+          track_span(span)
+        end
+
+        def example_group_finished(notification)
+          pop_and_finalize_span
+        end
+
+        def example_started(notification)
+          example = notification.example
+          attributes = {
+            'location' => example.location.to_s,
+            'full_description' => example.full_description.to_s,
+            'described_class' => example.metadata[:described_class].to_s
```

Comment:
I'd like to see `test.` prefix on the attribute names, or just some prefix if there's a more appropriate convention, not picky, just feels like a nice housekeeping thing to do while it's still greenfield

Reply:
Yes, I was considering `example.` or `rspec.` in order to make it specific to this test framework

I was thinking we should avoid `test.` in case it become part of the standard later and this library conflicts with it. Lot’s of speculation there anyway.

Reply:
yea, no strong opinions, `rspec` seems hard to argue with and future-proof enough for now?